components/datap_json/tests/test_file_io.rs

```diff
@@ -0,0 +1,32 @@
+extern crate icu_datap_json;
+
+use std::fs::File;
+use std::io::BufReader;
+
+use icu_datap_json::JsonDataProvider;
+use icu_util::datap;
+use icu_util::datap::DataProvider;
+
+#[test]
+fn test_read_json() {
+    let file = File::open("tests/testdata/all.json").unwrap();
+    let reader = BufReader::new(file);
+    let json_data_provider = JsonDataProvider::from_reader(reader).unwrap();
+    let response = json_data_provider
+        .load(datap::Request {
+            locale: "root".to_string(),
+            category: datap::Category::Decimal,
+            key: datap::Key::Decimal(datap::decimal::Key::SymbolsV1),
+            payload: None,
+        })
+        .unwrap();
+    let decimal_data: &datap::decimal::SymbolsV1 = response.borrow_payload().unwrap();
```

Comment:
Consider that versioning each data item separately is a major pain for the library users.

It may be easier on the users of the library to uprev at the API level granularity.  It's easier to change `use datap::decimal::v1` to `use datap::decimal::v2`. I agree that this is easier on the maintainer.

Reply:
The "library users" don't see these keys.  `use datap::decimal::v2` should be in only two places: ICU4X implementation code, and data provider code.